### PR TITLE
Add torch jit ignore decorator to Lightning properties

### PR DESCRIPTION
**Pull request by ananthsub**
## What does this PR do?

[This PyTorch PR](https://github.com/pytorch/pytorch/pull/42390) adds support to TorchScript for module properties, and as a result, these module properties in Lightning will no longer compile because they use unsupported features in their properties (that are now being compiled)

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

**Review by justusschock — APPROVED** [review]

**Comment by Borda** [comment]
@ananthsub seems all tests are failing, mind have a look?
`E   RuntimeError: Argument to @torch.jit.ignore must be a bool or a function but got <property object at 0x7f7506a384a8>`

**Comment by thomasjo** [comment]
@Borda The [upstream PR](https://github.com/pytorch/pytorch/pull/42390) hasn't been merged yet. This PR cannot be merged before the upstream PR is merged and available in a stable PyTorch release (possibly 1.7).

**Comment by Borda** [comment]
> @Borda The [upstream PR](https://github.com/pytorch/pytorch/pull/42390) hasn't been merged yet. This PR cannot be merged before the upstream PR is merged and available in a stable PyTorch release (possibly 1.7).

well, even 1.7 is out we need to maintain compatibility for pt<1.7 so we cannot merge it until this change is compatible also with older PT versions...

**Review by Borda — CHANGES_REQUESTED** [review]
we need to maintain compatibility with all supported PT versions, and this is valid only for the upcoming PT 1.7

**Comment by ananthsub** [comment]
this was handled by #3541

**Pull request closed without merging**
